File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.java
class Solution {
    Integer dp[];
    public int find(int i,int a[],int n){
        if(i>=n) return 0;
        if(i==n-1) return a[i];
        if(dp[i]!=null) return dp[i];
        return dp[i]=a[i]+Math.min(find(i+1,a,n),find(i+2,a,n));
    }
    public int minCostClimbingStairs(int[] cost) {
        dp=new Integer[cost.length+1];
        return Math.min(find(0,cost,cost.length),find(1,cost,cost.length));
        
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int min=int.MaxValue;
        int max=0;
        for(int i=0;i<prices.Length;i++){
            min=Math.Min(min,prices[i]);
            if(min<prices[i]){
                max=Math.Max(max,prices[i]-min);
            }
        }
        return max;
    }
}
